# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webgpu/CommandEncoder.cpp
# Commit: dd10bf41fcea
# Full Hash: dd10bf41fcea6aa2f40d4528f4a7afc22b1f389e
# Author: Nicolas Silva <nsilva@mozilla.com>
# Date: 2024-04-05 21:43:02
# Regressor Bug: 1881518
# File Overlap Count: 1
# Description:
#   Bug 1881518 - Recycle indices again. r=webgpu-reviewers,bradwerth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D202433
# ==============================================================================

diff -r b2cba928a7bb -r dd10bf41fcea dom/webgpu/CommandEncoder.cpp
--- a/dom/webgpu/CommandEncoder.cpp	Fri Apr 05 07:28:07 2024 +0000
+++ b/dom/webgpu/CommandEncoder.cpp	Fri Apr 05 07:42:05 2024 +0000
@@ -85,9 +85,16 @@
     return;
   }
   mValid = false;
+
+  if (!mBridge) {
+    return;
+  }
+
   if (mBridge->IsOpen()) {
     mBridge->SendCommandEncoderDrop(mId);
   }
+
+  wgpu_client_free_command_encoder_id(mBridge->GetClient(), mId);
 }
 
 void CommandEncoder::TrackPresentationContext(CanvasContext* aTargetContext) {